geode-lucene/src/test/java/org/apache/geode/cache/lucene/test/LuceneFunctionSecurityTest.java

```diff
@@ -0,0 +1,494 @@
+import org.apache.geode.cache.lucene.internal.directory.DumpDirectoryFiles;
+import org.apache.geode.cache.lucene.internal.distributed.LuceneQueryFunction;
+import org.apache.geode.cache.lucene.internal.distributed.WaitUntilFlushedFunction;
+import org.apache.geode.cache.lucene.internal.results.LuceneGetPageFunction;
+import org.apache.geode.cache.lucene.internal.security.LucenePermission;
+import org.apache.geode.examples.SimpleSecurityManager;
+import org.apache.geode.internal.logging.LogService;
+import org.apache.geode.security.ResourcePermission;
+import org.apache.geode.security.ResourcePermission.Operation;
+import org.apache.geode.security.ResourcePermission.Resource;
+import org.apache.geode.test.junit.categories.IntegrationTest;
+import org.apache.geode.test.junit.categories.SecurityTest;
+import org.apache.geode.test.junit.rules.ConnectionConfiguration;
+import org.apache.geode.test.junit.rules.GfshCommandRule;
+import org.apache.geode.test.junit.rules.ServerStarterRule;
+
+@Category({IntegrationTest.class, SecurityTest.class})
+public class LuceneFunctionSecurityTest {
+  private static final Logger logger = LogService.getLogger();
+
+  private static Function luceneCreateIndexFunction = Mockito.spy(new LuceneCreateIndexFunction());
+  private static Function luceneDescribeIndexFunction =
+      Mockito.spy(new LuceneDescribeIndexFunction());
+  private static Function luceneDestroyIndexFunction =
+      Mockito.spy(new LuceneDestroyIndexFunction());
+  private static Function luceneListIndexFunction = Mockito.spy(new LuceneListIndexFunction());
+  private static Function luceneSearchIndexFunction = Mockito.spy(new LuceneSearchIndexFunction());
+  private static Function dumpDirectoryFiles = Mockito.spy(new DumpDirectoryFiles());
+  private static Function luceneQueryFunction = Mockito.spy(new LuceneQueryFunction());
+  private static Function waitUntilFlushedFunction = Mockito.spy(new WaitUntilFlushedFunction());
+  private static Function luceneGetPageFunction = Mockito.spy(new LuceneGetPageFunction());
+
+  static {
+    Mockito.doAnswer(announceNoOp()).when(luceneCreateIndexFunction).execute(any());
+    Mockito.doAnswer(announceNoOp()).when(luceneDescribeIndexFunction).execute(any());
+    Mockito.doAnswer(announceNoOp()).when(luceneDestroyIndexFunction).execute(any());
+    Mockito.doAnswer(announceNoOp()).when(luceneListIndexFunction).execute(any());
+    Mockito.doAnswer(announceNoOp()).when(luceneSearchIndexFunction).execute(any());
+    Mockito.doAnswer(announceNoOp()).when(dumpDirectoryFiles).execute(any());
+    Mockito.doAnswer(announceNoOp()).when(luceneQueryFunction).execute(any());
+    Mockito.doAnswer(announceNoOp()).when(waitUntilFlushedFunction).execute(any());
+    Mockito.doAnswer(announceNoOp()).when(luceneGetPageFunction).execute(any());
+  }
+
+  private static String regionName = "luceneTestRegion";
+
+  private static ResourcePermission clusterManage =
```

Comment:
we should have this already defined in ResourcePermissions

Reply:
I forgot about that!

It's outside the scope of this ticket, but do we have any intention to extend `ResourcePermissions` or `LucenePermission` to define the finer-grained permissions?  We can't do it for permissions requiring a region name or key, but something like `DATA:WRITE:DISK` appears enough that it might be worth pre-defining.

Reply:
yeah, if needed, we can add there. it's minor though.